### PR TITLE
Best model saving and loading updates

### PR DESCRIPTION
**Pull request by gerazov**
Based on #643 this includes :
- best models names with a step stamp
- `best_model.pth.tar` symlink created for convinience
- saving multiple best models after a set step is available via `c.keep_best` and `c.keep_after`
- continuing training will find the checkpoint or the best model with the largest step stamp 
- when continuing training best loss is now taken from the latest best model

### CODE REVIEW

File: TTS/bin/train_glow_tts.py [hunk 1]
```diff
@@ -538,8 +538,16 @@ def main(args):  # pylint: disable=redefined-outer-name
...
+    if args.restore_step == 0 or not args.best_path:
         best_loss = float('inf')
+        print(" > Starting with inf best loss.")
+    else:
+        print(args.best_path)
+        best_loss = torch.load(args.best_path,
+                               map_location='cpu')['model_loss']
+        print(f" > Starting with loaded last best loss {best_loss}.")
+    keep_best = c.get('keep_best', False)
+    keep_after = c.get('keep_after', 10000)  # void if keep_best False
```
Comment (erogol) [inline TTS/bin/train_glow_tts.py]: What do keep_best and keep_after do?

Comment (gerazov) [reply TTS/bin/train_glow_tts.py]: I've commented it in the configs. Basically `keep_best` keeps all the best models if `current_step > keep_after`.

File: TTS/utils/arguments.py [hunk 2]
```diff
@@ -67,11 +72,11 @@ def parse_arguments(argv):
     return parser.parse_args()
 
 
-def get_last_checkpoint(path):
-    """Get latest checkpoint from a list of filenames.
+def get_last_models(path):
+    """Get latest checkpoint or/and best model in path.
```
Comment (erogol) [inline TTS/utils/arguments.py:72]: For the docstrings we were using https://sphinxcontrib-napoleon.readthedocs.io/en/latest/example_google.html

Comment (gerazov) [reply TTS/utils/arguments.py:72]: It's more compact :+1: These are the numpy ones that Spyder automatically generates. I'll redo them.

Comment (erogol) [reply TTS/utils/arguments.py:72]: sry for the rework :(

Comment (gerazov) [reply TTS/utils/arguments.py:72]: As a side note - are the circleci test runs free? If not maybe we can only do them once the PR is complete?

File: TTS/tts/configs/ljspeech_tacotron2_dynamic_conv_attn.json [hunk 3]
```diff
@@ -1,171 +1,173 @@
...
-            {
-                "name": "ljspeech",
-                "path": "/home/erogol/Data/LJSpeech-1.1/",
-                "meta_file_train": "metadata.csv", // for vtck if list, ignore speakers id in list for train, its useful for test cloning with new speakers
-                "meta_file_val": null
-            }
-        ]
-}
-
+{
```
Comment (erogol) [inline TTS/tts/configs/ljspeech_tacotron2_dynamic_conv_attn.json:1]: did you change anything here?

File: TTS/vocoder/configs/multiband-melgan_and_rwd_config.json [hunk 4]
```diff
@@ -138,6 +138,8 @@
     "print_eval": false,     // If True, it prints loss values for each step in eval run.
     "save_step": 25000,      // Number of training steps expected to plot training stats on TB and save model checkpoints.
     "checkpoint": true,     // If true, it saves checkpoints per "save_step"
+    "keep_best": false,  // If true, keeps all best_models after keep_after steps
```
Comment (erogol) [inline TTS/vocoder/configs/multiband-melgan_and_rwd_config.json]: Right now what happens if keep_best: false? Does it keep any best_model at all in that case?

Comment (gerazov) [reply TTS/vocoder/configs/multiband-melgan_and_rwd_config.json]: It will keep just the last best model deleting previous ones. It's a bit misleading you're right - maybe we change it to `keep_all_best`?

File: TTS/tts/configs/ljspeech_tacotron2_dynamic_conv_attn.json (hunk 3 above)
Comment (gerazov) [reply TTS/tts/configs/ljspeech_tacotron2_dynamic_conv_attn.json:1]: Wow :confused: I just added the same two lines in all the configs. I'm not sure what went wrong here ... Does GitHub allow side-by-side comparison to check just in case? GitLab has nice support for this...

Another side note - `json` not allowing comments is a bit frustrating, i.e. syntax coloring goes berserk in some editors (vim) and even here in GitHub, maybe at some point (in the far future) we transition to something more permissive :slightly_smiling_face:

File: TTS/utils/arguments.py (hunk 2 above)
Comment (erogol) [reply TTS/utils/arguments.py:72]: It is not something I'm worried about so far :) but thx for the flag.

File: TTS/vocoder/configs/multiband-melgan_and_rwd_config.json (hunk 4 above)
Comment (erogol) [reply TTS/vocoder/configs/multiband-melgan_and_rwd_config.json]: I think ```keep_all_best``` is more intuitive.

File: TTS/utils/arguments.py (hunk 2 above)
Comment (erogol) [reply TTS/utils/arguments.py:72]: I think keeping the function name as before is more intuitive. In the end best model is also a checkpoint.

**Comment by gerazov** [comment]
Ok this is good to go. We could also do a `keep_all_checkpoints` to complement the keeping the best model logic, but only if it sounds good :slightly_smiling_face: and maybe in a different PR ...
